src/window/gui/Gui.cpp

```diff
@@ -232,6 +232,11 @@ void Gui::ImGuiBackendInit() {
 void Gui::LoadTextureFromRawImage(const std::string& name, const std::string& path) {
     const auto res = Context::GetInstance()->GetResourceManager()->GetArchiveManager()->LoadFileRaw(path);
 
+    if (!res || !res->Buffer || res->Buffer->empty()) {
```

Comment:
Can you separate out the buffer check from the res == null check?

Reply:
> Can you separate out the buffer check from the res == null check?

Yeh, no problem. Check out the latest commit.